xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -435,39 +436,48 @@ def StencilConversion(return_targets: dict[ReturnOp, CastOp | memref.Cast],
     ])
 
 
-# TODO: We probably want to factor that in another file
-def StencilShapeInference(ctx: MLContext, module: ModuleOp):
+class StencilShapeInferencePass(ModulePass):
+
+    name = 'stencil-shape-inference'
+
+    inference_walker = PatternRewriteWalker(ShapeInference,
```

Comment:
I generally get a weird feeling when defining fields/functions with default objects in Python, so I would generally prefer to move this into the apply function.

In a later PR, we might also want to integrate this more deeply with our current rewriting infrastructure, s.t. you could for example override the `walker` field, or a `Pattern` list and then you don't have to even give an apply method. Let's leave this for later though :)

Reply:
Sure, leaving things out of apply was more a style thing to me, I can leave everything in for now.

+10 for the deeper integration ☺️